dateutil/parser/_parser.py

```diff
@@ -571,6 +571,74 @@ def resolve_ymd(self, yearfirst, dayfirst):
         return year, month, day
 
 
+def next_jump(tokens, idx, skip_comma=True):
```

Comment:
I think this should be a method somewhere, no?

I definitely think it shouldn't be a public, top-level function yet. Can you move it onto either `parser` or `parserinfo` (whichever you think is best), and start it out as private (`_next_jump`)?

Reply:
> I think this should be a method somewhere, no? [...] Can you move it onto either `parser` or `parserinfo`

Downstream I actually have these as methods in a `TokenStream` class (subclasses `list`) and `tokens = TokenStream(timestr)` takes the place of `l = _timelex(timestr)`.  In #884 I'll describe some of the other virtues of the `TokenStream` class.  If this is made into a method, I think `TokenStream` is the right abstraction for where it belongs.

For the time being I've privatized the function.